src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTreeExtensions.cs

```diff
@@ -73,32 +73,30 @@ public static ISet<SyntaxKind> GetPrecedingModifiers(this SyntaxTree syntaxTree,
             return result;
         }
 
-        public static TypeDeclarationSyntax? GetContainingTypeDeclaration(
+        public static BaseTypeDeclarationSyntax? GetContainingTypeDeclaration(
             this SyntaxTree syntaxTree, int position, CancellationToken cancellationToken)
         {
             return syntaxTree.GetContainingTypeDeclarations(position, cancellationToken).FirstOrDefault();
         }
 
-        public static BaseTypeDeclarationSyntax? GetContainingTypeOrEnumDeclaration(
-            this SyntaxTree syntaxTree, int position, CancellationToken cancellationToken)
-        {
-            return syntaxTree.GetContainingTypeOrEnumDeclarations(position, cancellationToken).FirstOrDefault();
-        }
-
-        public static IEnumerable<TypeDeclarationSyntax> GetContainingTypeDeclarations(
+        public static IEnumerable<BaseTypeDeclarationSyntax> GetContainingTypeDeclarations(
             this SyntaxTree syntaxTree, int position, CancellationToken cancellationToken)
         {
             var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
 
-            return token.GetAncestors<TypeDeclarationSyntax>().Where(t =>
-            {
-                return BaseTypeDeclarationContainsPosition(t, position);
-            });
+            return token.GetAncestors<BaseTypeDeclarationSyntax>().Where(t => BaseTypeDeclarationContainsPosition(t, position));
         }
 
         private static bool BaseTypeDeclarationContainsPosition(BaseTypeDeclarationSyntax declaration, int position)
         {
-            if (position <= declaration.OpenBraceToken.SpanStart)
+            if (declaration is TypeDeclarationSyntax typeDeclaration &&
+                position <= typeDeclaration.Keyword.SpanStart)
+            {
+                return false;
+            }
+
+            if (declaration is EnumDeclarationSyntax enumDeclaration &&
+                position <= enumDeclaration.EnumKeyword.SpanStart)
```

Comment:
why this new logic?  this is actually not what i would expect and i think we should revert this.  it was very intentional the distinction between being *on* versus *inside*.  So if you're on the thing, you're not contained *inside*  it.